### PR TITLE
Update notebooks

### PR DESCRIPTION
**Pull request by jwallwork23**
Ahead of the next Firedrake training, I went through all of the notebooks and updated to modern Firedrake.

Included updates:
* PETSc output is now captured by Firedrake and appears in the notebooks; I reduced the verbosity in some cases.
* LU solver defaults mean that a cell which was meant to break in the elasticity notebook was working fine.
* LU solver defaults mean that a cell which was meant to work in mixed Poisson was broken.
* Added a `get_solve_blocks` helper function which is used in the adjoint notebook.
* Link to adjoint notebook from docs.
* Formatting, typos and punctuation.

I ran all of the notebooks using a newly updated Firedrake.

### CODE REVIEW

**Comment by jwallwork23** [comment]
Running locally, I get an error in cell 10 of the sum factorisation notebook:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-205a191d5f9c> in <module>
----> 1 gll_quadrature_rule = gauss_lobatto_legendre_cube_rule(dimension=3, degree=p)
      2 a_gll = dot(grad(u), grad(v)) *dx(rule=gll_quadrature_rule)

<ipython-input-8-7c4d435c3d14> in gauss_lobatto_legendre_cube_rule(dimension, degree)
     10 def gauss_lobatto_legendre_cube_rule(dimension, degree):
     11     make_tensor_rule = finat.quadrature.TensorProductQuadratureRule
---> 12     result = gauss_lobatto_legendre_line_rule(degree)
     13     for _ in range(1, dimension):
     14         line_rule = gauss_lobatto_legendre_line_rule(degree)

<ipython-input-8-7c4d435c3d14> in gauss_lobatto_legendre_line_rule(degree)
      6     finat_ps = finat.point_set.GaussLobattoLegendrePointSet
      7     finat_qr = finat.quadrature.make_quadrature
----> 8     return finat_qr(finat_ps(fiat_rule.get_points()), fiat_rule.get_weights())
      9 
     10 def gauss_lobatto_legendre_cube_rule(dimension, degree):

~/software/firedrake/src/FInAT/finat/quadrature.py in make_quadrature(ref_el, degree, scheme)
     28         integrate exactly.
     29     """
---> 30     if ref_el.get_shape() == TENSORPRODUCT:
     31         try:
     32             degree = tuple(degree)

AttributeError: 'GaussLobattoLegendrePointSet' object has no attribute 'get_shape'
```
Can anyone reproduce it?

I don't know enough about FInAT to recommend a fix.

**Comment by jwallwork23** [comment]
Where is the code that tells the test suite to run the notebooks? I can't see it. Notebooks 5 and 10 were broken before this PR and notebook 4 was meant to fail but didn't, so it seems like something has been going wrong in testing.

File: docs/notebooks/10-sum-factorisation.ipynb [hunk 1]
```diff
@@ -219,8 +219,10 @@
     "    fiat_make_rule = FIAT.quadrature.GaussLobattoLegendreQuadratureLineRule\n",
     "    fiat_rule = fiat_make_rule(FIAT.ufc_simplex(1), degree + 1)\n",
     "    finat_ps = finat.point_set.GaussLobattoLegendrePointSet\n",
-    "    finat_qr = finat.quadrature.make_quadrature\n",
-    "    return finat_qr(finat_ps(fiat_rule.get_points()), fiat_rule.get_weights())\n",
+    "    points = finat_ps(fiat_rule.get_points())\n",
+    "    weights = fiat_rule.get_weights()\n",
+    "    finat_qr = finat.quadrature.QuadratureRule\n",
+    "    return finat_qr(points, weights)\n",
```
Comment (dham) [inline docs/notebooks/10-sum-factorisation.ipynb:225]: Are we adding anything by introducing the variable `finat_qr`? Maybe we should just use `finat.quadrature.QuadratureRule` directly.

**Review by dham — CHANGES_REQUESTED** [review]
Thanks for this, this is a needed revision. I just had one tiny suggestion.

File: docs/notebooks/10-sum-factorisation.ipynb (hunk 1 above)
Comment (jwallwork23) [reply docs/notebooks/10-sum-factorisation.ipynb:225]: I guess not, will change now.

**Review by dham — APPROVED** [review]

**Pull request merged**